src/utils/saml.ts

```diff
@@ -564,26 +609,37 @@ export const getPreValidateResponse = (
       TE.map(() => _)
     );
 
+  /* LOGGING INFOS:
+    having the idpIssuer and requestId as data here we leverage multiple advantages:
+    1. we can query based on the idp and display graphs about errors/usage
+    2. we know what idp is causing the error
+    3. having the requestId it's possible to analyze further the problem encountered
+  */
   const validationFailure = (error: Error | ITransformValidation): void => {
     if (eventHandler) {
-      // eslint-disable-next-line @typescript-eslint/no-unused-expressions
-      TransformError.is(error)
-        ? eventHandler({
-            data: {
-              idpIssuer: error.idpIssuer,
-              message: error.message,
-              numberOfTransforms: String(error.numberOfTransforms)
-            },
-            name: "spid.error.transformOccurenceOverflow",
-            type: "ERROR"
-          })
-        : eventHandler({
-            data: {
-              message: error.message
-            },
-            name: "spid.error.generic",
-            type: "ERROR"
-          });
+      if (TransformError.is(error)) {
+        eventHandler({
+          data: {
+            idpIssuer: error.idpIssuer,
+            message: error.message,
+            numberOfTransforms: String(error.numberOfTransforms),
+            requestId
+          },
+          name: "spid.error.transformOccurenceOverflow",
+          type: "ERROR"
+        });
+      } else {
+        const samlError: ISAMLError = toSamlError(error.message);
+        eventHandler({
+          data: {
+            idpIssuer: samlError.idpIssuer,
+            message: samlError.message,
+            requestId: samlError.requestId
```

Comment:
the `toSamlError` function is used only here. What is his purpose?
Will be the same this version
```suggestion
        const samlError: ISAMLError = toSamlError(error.message);
        eventHandler({
          data: {
            idpIssuer,
            message: error.message,
            requestId
```

Reply:
fixed in https://github.com/pagopa/io-spid-commons/pull/121/commits/8a663a2b24774f521a9a08720d21d5a348d7a998